### PR TITLE
Fix gob encoder error on unknown value

### PR DESCRIPTION
**Pull request by syndicut**
When using Config() method with `--module` option you can get cty.unknownType inside module variables without default values and following error:
```
13:39:46 server.go:418: rpc: gob error encoding body: error encoding cty.Value: gob: type not registered for interface: cty.unknownType
```
This fix forces cty.Value{} in case of unknown value

### CODE REVIEW

**Comment by wata727** [comment]
Umm, I understand this bug, but this fix makes it impossible for the plugin to handle variables as unknown.

Maybe gob is not a good way to transfer `cty.Value`. go-cty seems to support encoding/decoding with msgpack, so I think this might be a good idea, but I haven't started investigating anything yet. If you are interested, I would like to support you as much as possible.
https://pkg.go.dev/github.com/zclconf/go-cty@v1.7.1/cty/msgpack

**Comment by syndicut** [comment]
> Umm, I understand this bug, but this fix makes it impossible for plugins to handle variables as unknown.
> 
> Maybe gob is not a good way to transfer `cty.Value`. go-cty seems to support encoding/decoding with msgpack, so I think this might be a good idea, but I haven't started investigating anything yet. If you are interested, I would like to support you as much as possible.
> https://pkg.go.dev/github.com/zclconf/go-cty@v1.7.1/cty/msgpack

Ok, good point, will look at it

**Comment by syndicut** [comment]
> Umm, I understand this bug, but this fix makes it impossible for plugins to handle variables as unknown.
> 
> Maybe gob is not a good way to transfer `cty.Value`. go-cty seems to support encoding/decoding with msgpack, so I think this might be a good idea, but I haven't started investigating anything yet. If you are interested, I would like to support you as much as possible.
> https://pkg.go.dev/github.com/zclconf/go-cty@v1.7.1/cty/msgpack

Think those changes and terraform-linters/tflint-plugin-sdk#96 should do the trick

**Comment by wata727** [comment]
Merged https://github.com/terraform-linters/tflint-plugin-sdk/pull/96. Could you update the sdk version in this PR?

**Comment by syndicut** [comment]
> Merged [terraform-linters/tflint-plugin-sdk#96](https://github.com/terraform-linters/tflint-plugin-sdk/pull/96). Could you update the sdk version in this PR?

Done!

File: plugin/encode.go [hunk 1]
```diff
@@ -360,11 +362,14 @@ func (s *Server) encodeVariable(variable *tfconfigs.Variable) *tfplugin.Variable
 		validations[i] = s.encodeVariableValidation(v)
 	}
 
+	defaultVal, _ := msgpack.Marshal(variable.Default, variable.Default.Type())
+	// We need to use json because cty.DynamicPseudoType cannot be encoded by gob
+	typeVal, _ := json.MarshalType(variable.Type)
```
Comment (wata727) [inline plugin/encode.go:367]: We may need to extend `encodeVariable` to properly return this error to the caller.

**Review by wata727 — APPROVED** [review]
👍

**Pull request merged**